### PR TITLE
Added support for long-term statistics

### PR DESCRIPTION
**Pull request by MartinSchmidt**
This PR adds a new sensor `sensor.eloverblik_energy_statistic`, this sensor uses `async_import_statistics` functionality to import long-term statistics into HomeAssistant.

The statistics are inserted at the correct point in time and amount, also tested around summer/winter time. 

The sensor always has an unknown value as to not make the recorder log the value, but one can still use its long-term statistics.

I have extended the readme with example on how to use it on a dashboard.

### CODE REVIEW

**Comment by MartinSchmidt** [comment]
Just discovered another small bug (a missing if statement) , I will get it fix one of the following days.

**Comment by MartinSchmidt** [comment]
I fixed the bug, and has been running for the better part of a week without issues

@JonasPed  will you take a look at this PR 🙂

**Comment by JonasPed** [comment]
Thanks. I will take a look at it and test it as soon as I have a few hours of spare time.

**Comment by frenzeldk** [comment]
After reading through the changes and applying the patch to my installation and having it run for a while I have not noticed any issues. For obvious reasons I have not tested the summer/winter time functionality.

**Review by JonasPed — CHANGES_REQUESTED** [review]
Look very good. Just a couple of minor comments. 

Would it not be an idea to hide the statistics sensor by default? Hidden sensors is still updated they are just not shown in the UI by default. I think this can be done by adding a property named entity_registry_visible_default and let it return False. This way we will hide the sensor that always reads Unknown.

**Comment by JonasPed** [comment]
It have been running in a test setup for a couple of days now and every night/morning I see something like below. Will turn on some debugging to get further details. 

```
2023-01-31 05:35:56.647 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/config/custom_components/eloverblik/sensor.py", line 206, in _update_data
    await self._insert_statistics(data, last_stat)
  File "/config/custom_components/eloverblik/sensor.py", line 226, in _insert_statistics
    number_of_hours = len(time_series._metering_data)
TypeError: object of type 'NoneType' has no len()

```

**Comment by MartinSchmidt** [comment]
I will have a look at it during the weekend, and also the vommenes you wrote.

File: README.md [hunk 1]
```diff
@@ -160,3 +143,64 @@ template:
...
+this entity will **always** have an `unknown` value, since if a current value is set, 
+the recorder will try to write it to the statistics. 
+
+> **_NOTE:_**  This is not the ideal setup, but it does enable owners that do not have live access to their measurements to get the data into home-assitant.
+
+But the entity will have a valid long-term statistic.
+
+The statistic will continually be updateed daily.
+
+> **_NOTE:_**  The data will be between 1 and 3 days of delay, depending on your DSO.
```
[the remark this answers is not in the document]
Comment (MartinSchmidt) [reply README.md]: I have changed the description, hope it is more understandable.

File: custom_components/eloverblik/manifest.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "domain": "eloverblik",
   "name": "Eloverblik",
-  "version": "0.2.0",
+  "version": "0.5.0",
```
[the remark this answers is not in the document]
Comment (MartinSchmidt) [reply custom_components/eloverblik/manifest.json]: I have reverted the version, i though it was supposed to follow the versions you create her on github.

**Comment by MartinSchmidt** [comment]
> It have been running in a test setup for a couple of days now and every night/morning I see something like below. Will turn on some debugging to get further details.
> 
> ```
> 2023-01-31 05:35:56.647 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
> Traceback (most recent call last):
>   File "/config/custom_components/eloverblik/sensor.py", line 206, in _update_data
>     await self._insert_statistics(data, last_stat)
>   File "/config/custom_components/eloverblik/sensor.py", line 226, in _insert_statistics
>     number_of_hours = len(time_series._metering_data)
> TypeError: object of type 'NoneType' has no len()
> ```

I have added some extra validation, it should no longer throw the error, but I will keep an eye on my local instance the next couple of days.

**Comment by JonasPed** [comment]
Looks great. I will update my own installation and let it run for a few days before merging.

**Comment by JonasPed** [comment]
Thanks

**Pull request merged**